### ISSUE TITLE
consumer端抛不出原始异常

### ISSUE THREAD

**Issue by a912327347** [apache/incubator-dubbo#3705]
Environment
Dubbo version: 2.7.0
Operating System version: windows 10
Java version: 1.8
Steps to reproduce this issue
org.apache.dubbo.config.spring.beans.factory.annotation.ReferenceAnnotationBeanPostProcessor中
ReferenceBean对象的get()返回的已经是org.apache.dubbo.rpc.ProxyFactory创建的代理对象为什么还需要再次代理？ 

以前版本没有被再次代理时consumer端异常抛出来是包装原始异常的InvocationTargetException  
现在再次代理后异常变成了包装UndeclaredThrowableException异常的InvocationTargetException

![image](https://user-images.githubusercontent.com/8800733/54683870-bf909780-4b4d-11e9-9a3a-86b312fb46df.png)

**Comment by cvictory** [comment]
I write a demo and test in dubbo-2.7.0.  The exception log:

```
java.lang.reflect.UndeclaredThrowableException
	at com.sun.proxy.$Proxy25.sayGoodbye(Unknown Source)
	at org.apache.dubbo.samples.annotation.action.AnnotationAction.doSayGoodbye(AnnotationAction.java:58)
	at org.apache.dubbo.samples.annotation.AnnotationConsumerBootstrap.main(AnnotationConsumerBootstrap.java:41)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.dubbo.config.spring.beans.factory.annotation.ReferenceAnnotationBeanPostProcessor$ReferenceBeanInvocationHandler.invoke(ReferenceAnnotationBeanPostProcessor.java:158)
	... 3 more
Caused by: java.lang.RuntimeException: sss
	at org.apache.dubbo.samples.annotation.impl.AnnotationHelloServiceImpl.sayGoodbye(AnnotationHelloServiceImpl.java:40)
	at org.apache.dubbo.common.bytecode.Wrapper3.invokeMethod(Wrapper3.java)
	at org.apache.dubbo.rpc.proxy.javassist.JavassistProxyFactory$1.doInvoke(JavassistProxyFactory.java:47)
	at org.apache.dubbo.rpc.proxy.AbstractProxyInvoker.invoke(AbstractProxyInvoker.java:86)
	at org.apache.dubbo.config.invoker.DelegateProviderMetaDataInvoker.invoke(DelegateProviderMetaDataInvoker.java:56)
	at org.apache.dubbo.rpc.protocol.InvokerWrapper.invoke(InvokerWrapper.java:56)
	at org.apache.dubbo.rpc.filter.ExceptionFilter.invoke(ExceptionFilter.java:63)
	at org.apache.dubbo.rpc.protocol.ProtocolFilterWrapper$1.invoke(ProtocolFilterWrapper.java:73)
	at org.apache.dubbo.monitor.support.MonitorFilter.invoke(MonitorFilter.java:88)
	at org.apache.dubbo.rpc.protocol.ProtocolFilterWrapper$1.invoke(ProtocolFilterWrapper.java:73)
	at org.apache.dubbo.rpc.filter.TimeoutFilter.invoke(TimeoutFilter.java:42)
	at org.apache.dubbo.rpc.protocol.ProtocolFilterWrapper$1.invoke(ProtocolFilterWrapper.java:73)
	at org.apache.dubbo.rpc.protocol.dubbo.filter.TraceFilter.invoke(TraceFilter.java:79)
	at org.apache.dubbo.rpc.protocol.ProtocolFilterWrapper$1.invoke(ProtocolFilterWrapper.java:73)
	at org.apache.dubbo.rpc.filter.ContextFilter.invoke(ContextFilter.java:78)
	at org.apache.dubbo.rpc.protocol.ProtocolFilterWrapper$1.invoke(ProtocolFilterWrapper.java:73)
	at org.apache.dubbo.rpc.filter.GenericFilter.invoke(GenericFilter.java:138)
	at org.apache.dubbo.rpc.protocol.ProtocolFilterWrapper$1.invoke(ProtocolFilterWrapper.java:73)
	at org.apache.dubbo.rpc.filter.ClassLoaderFilter.invoke(ClassLoaderFilter.java:38)
	at org.apache.dubbo.rpc.protocol.ProtocolFilterWrapper$1.invoke(ProtocolFilterWrapper.java:73)
	at org.apache.dubbo.rpc.filter.EchoFilter.invoke(EchoFilter.java:39)
	at org.apache.dubbo.rpc.protocol.ProtocolFilterWrapper$1.invoke(ProtocolFilterWrapper.java:73)
	at org.apache.dubbo.rpc.protocol.dubbo.DubboProtocol$1.reply(DubboProtocol.java:108)
	at org.apache.dubbo.remoting.exchange.support.header.HeaderExchangeHandler.handleRequest(HeaderExchangeHandler.java:103)
	at org.apache.dubbo.remoting.exchange.support.header.HeaderExchangeHandler.received(HeaderExchangeHandler.java:200)
	at org.apache.dubbo.remoting.transport.DecodeHandler.received(DecodeHandler.java:51)
	at org.apache.dubbo.remoting.transport.dispatcher.ChannelEventRunnable.run(ChannelEventRunnable.java:57)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by chickenlj** [comment]
We should throw `Caused by: java.lang.RuntimeException: sss` directly to the consumer.

**Comment by chickenlj** [comment]
> 以前版本没有被再次代理时consumer端异常抛出来是包装原始异常的InvocationTargetException
现在再次代理后异常变成了包装UndeclaredThrowableException异常的InvocationTargetException

I can understand why `UndeclaredThrowableException` was thrown after being proxied two times. 

But I don't understand why `InvocationTargetException` was thrown when there's only one proxy, are you sure about that? I think the original `java.lang.RuntimeException: sss` should be thrown in this case.

**Comment by cvictory** [comment]
I agree with the publisher. 
I don't know why we have two proxies, one is from ReferenceBean.get, and another is from InvocationHandler.

Can we remove ReferenceBeanInvocationHandler and Proxy.newProxyInstance?  We just use ReferenceBean.get directly.

I will try to work on this.

**Comment by cvictory** [comment]
It is fixed in dubbo-2.7.1, and it will be released later.

https://github.com/apache/incubator-dubbo/blob/2.7.1-release/dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/beans/factory/annotation/ReferenceAnnotationBeanPostProcessor.java#L166

**Comment by kele6413430** [comment]
> It is fixed in dubbo-2.7.1, and it will be released later.
> 
> https://github.com/apache/incubator-dubbo/blob/2.7.1-release/dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/beans/factory/annotation/ReferenceAnnotationBeanPostProcessor.java#L166

should we merge it in 2.7.0-release ? 2.7.1 has other bugs,so I can not use 2.7.1

**Comment by a949690645** [comment]
Good job,bro!
